lib/ansible/modules/unarchive.py

```diff
@@ -1054,6 +1054,9 @@ def main():
     if not os.access(src, os.R_OK):
         module.fail_json(msg="Source '%s' not readable" % src)
 
+    # ensure src is an absolute path before picking handlers
+    src = os.path.abspath(src)
```

Comment:
Would it make sense/be possible to fix this at a higher level so other modules are not impacted?

Reply:
This is what I thought at first, but I wasn't sure how to best do that. I thought maybe it would require another remote call in `_transfer_file` after the transfer to locate the absolute destination, because _transfer_file https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/action/unarchive.py#L87-L88 but just returns the destination passed in. The connection plugin's `put_file` runs the command to transfer the src but doesn't determine an absolute path either.

Or, this seems like a better approach, maybe type='path'` could just handle this in the argspec. I don't know whether that might reasonably cause issues for things relying on the current behavior, do you think something like that should be opt-in?

I think modules usually specify document/handle whether remote/absolute paths are supported on a case-by-case basis, so I'm not actually sure this is a general issue. This issue also occurs with a remote relative src, the tmp dir isn't a necessary factor. The unarchive documentation for src is a little ambiguous to me:

>  If remote_src=no (default), local path to archive file to copy to the target server; can be absolute or relative. If remote_src=yes, path on the target server to existing archive file to unpack.

I read that as a controller-sourced `src`can definitely be absolute or relative, but it's less clear that a remote src can only be absolute (and I don't see why it should be).

I could look into adding this to the arspec if you think that would  be better, or have other suggestions.